src/jit/SimdArm64Inl.h

```diff
@@ -39,6 +59,53 @@ static void simdEmitOp(sljit_compiler* compiler, uint32_t opcode, sljit_s32 rd,
     sljit_emit_op_custom(compiler, &opcode, sizeof(uint32_t));
 }
 
+static void simdEmitI64x2Mul(sljit_compiler* compiler, sljit_s32 rd, sljit_s32 rn, sljit_s32 rm)
+{
+    auto tmpReg1 = SLJIT_FR2;
+    auto tmpReg2 = SLJIT_FR3;
+
+    uint32_t opcode = SimdOp::rev64;
+    opcode |= SimdOp::S4;
+
+    simdEmitOp(compiler, opcode, tmpReg2, rm, 0);
+    simdEmitOp(compiler, SimdOp::mul | SimdOp::S4, tmpReg2, tmpReg2, rn);
+
+    opcode = SimdOp::xtn;
+    opcode |= SimdOp::S4;
+    simdEmitOp(compiler, opcode, tmpReg1, rn, 0);
+
+    opcode = SimdOp::addp;
+    opcode |= SimdOp::S4;
+    simdEmitOp(compiler, opcode, rd, tmpReg2, tmpReg2);
+
+    opcode = SimdOp::xtn;
+    opcode |= SimdOp::S4;
+    simdEmitOp(compiler, opcode, tmpReg2, rm, 0);
+
+    opcode = SimdOp::shll;
+    opcode |= SimdOp::S4;
+    simdEmitOp(compiler, opcode, rd, rd, 0);
+
+    opcode = SimdOp::umlal;
+    opcode |= SimdOp::S4;
+    simdEmitOp(compiler, opcode, rd, tmpReg2, tmpReg1);
+}
+
+static void simdEmitSaturatedAddOrSub(sljit_compiler* compiler, sljit_s32 rd, sljit_s32 rn, sljit_s32 rm, SimdOp::SizeType size, bool isAdd, bool isSigned)
+{
+    uint32_t opcode = SimdOp::sqadd;
```

Comment:
these are the instruction forms: squadd / uqadd / sqsub / uqsub
if there is a problem with them, it is easier to find them in the documentation